virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -224,24 +239,68 @@ def setup(self) -> None:
             )
         )
 
+        # Turn elevation data into np array
+        elevation = np.array(self.data["elevation"])
+
+        # map neighbours and identify downstream cell_ids
+        mapping = [
+            self.data.grid.neighbours[cell_id]
+            for cell_id in range(len(self.data.grid.cell_id))
+        ]
+        downstream_ids = []
+
+        for cell_id, neighbors_id in enumerate(mapping):
+            # Find lowest neighbour
+            downstream_id_loc = np.argmax(elevation[cell_id] - elevation[neighbors_id])
+            downstream_ids.append(neighbors_id[downstream_id_loc])
+
+        # Invert mapping of cell_id: downstream neighbour to cell_id: upstream_neighbors
+        upstream_ids: list = [[] for i in range(len(self.data.grid.cell_id))]
+        for down_s, up_s in enumerate(downstream_ids):
+            upstream_ids[up_s].append(down_s)
+
+        self.upstream_ids = upstream_ids
+        """Upstream ID's for the calculation of accumulated surface runoff."""
+
+        # Get the runoff created by SPLASH or initial data set as the baseline:
+        baseline_runoff = np.array(self.data["surface_runoff"])
```

Comment:
The hydrology model needs some initial surface runoff to calculate the accumulated runoff in the setup because the update needs an input from the previous time step. At the moment, surface runoff is calculated in the hydrology model at each time step. If we implement SPLASH, the surface runoff will be provided and the calculation in the hydrology model can be removed/ made optional.
As far as I understand, the SPLASH model will be part of the pyrealm package but still somewhat separate from the plant model.

Reply:
That is correct. The intention is for `pyrealm` to provide a modern Python implementation of SPLASH v1.0, which could be then be imported and used in one of the `abiotic` models.